src/instruments/src/SD/Pages/Eng/Eng.tsx

```diff
@@ -0,0 +1,312 @@
+    const OIL_PSI_HIGH_LIMIT = 130;
+    const OIL_PSI_LOW_LIMIT = 14; // TODO FIXME: standin value
+    const OIL_PSI_VLOW_LIMIT = 12;
+    const [psiNeedleRed, setPsiNeedleRed] = useState(true);
+    const [pressureAboveHigh, setPressureAboveHigh] = useState(false);
+    const [pressureBelowLow, setPressureBelowLow] = useState(false);
+
+    const [engineOilTemperature] = useSimVar(`GENERAL ENG OIL TEMPERATURE:${engineNumber}`, 'celsius', 250);
+    const OIL_TEMP_HIGH_ADVISORY = 140;
+    const OIL_TEMP_VHIGH_LIMIT = 155;
+    const displayedEngineOilTemperature = Math.round(engineOilTemperature / 5) * 5; // Engine oil temperature has a step of 5
+    const [tempAmber, setTempAmber] = useState(false);
+    const [shouldTemperaturePulse, setShouldTemperaturePulse] = useState(false);
+    const [tempBeenAboveAdvisory, setTempBeenAboveAdvisory] = useState(false);
+
+    const [n1Vibration] = useSimVar(`TURB ENG VIBRATION:${engineNumber}`, 'Number');
+
+    const [n2Vibration] = useSimVar(`TURB ENG VIBRATION:${engineNumber}`, 'Number'); // FIXME TODO: should have a different value than N1, currently API limited
+
+    function getNeedleValue(value: any, max: number): number {
+        const numberValue = (Number)(value);
+        if (numberValue < max) {
+            return (numberValue / max) * 100;
+        }
+        return 100;
+    }
+
+    useEffect(() => {
+        if (flightPhase === 2 || flightPhase === 6) {
+            if (displayedEngineOilPressure > OIL_PSI_HIGH_LIMIT - 1) {
+                setPressureAboveHigh(true);
+            }
+
+            if (pressureAboveHigh && displayedEngineOilPressure < OIL_PSI_HIGH_LIMIT - 4) {
+                setPressureAboveHigh(false);
+            }
+
+            if (displayedEngineOilPressure < OIL_PSI_LOW_LIMIT && n2Percent > 75) {
+                setPressureBelowLow(true);
+            }
+
+            if (pressureBelowLow && displayedEngineOilPressure > OIL_PSI_LOW_LIMIT + 2) {
+                setPressureBelowLow(true);
+            }
+
+            if (pressureAboveHigh || pressureBelowLow) {
+                setShouldPressurePulse(true);
+            } else {
+                setShouldPressurePulse(false);
+            }
+
+            if (displayedEngineOilQuantity <= OIL_QTY_LOW_ADVISORY) {
+                setQuantityAtOrBelowLow(true);
+            }
+
+            if (quantityAtOrBelowLow && displayedEngineOilQuantity >= OIL_QTY_LOW_ADVISORY + 2) {
+                setQuantityAtOrBelowLow(false);
+            }
+        } else {
+            setShouldPressurePulse(false);
+            setShouldQuantityPulse(false);
+        }
+
+        if (displayedEngineOilPressure <= OIL_PSI_VLOW_LIMIT) {
+            setPsiNeedleRed(true);
+        }
+        if (psiNeedleRed && displayedEngineOilPressure >= OIL_PSI_VLOW_LIMIT + 0.5) {
+            setPsiNeedleRed(false);
+        }
+    }, [flightPhase, engineOilQuantity, engineOilPressure]);
+
+    useEffect(() => {
```

Comment:
**issue:** This `useEffect` is long. It should have a comment above describing what effects it is handling.